### DISCUSSION TITLE
Using sliding_window_inference only on specific regions

### DISCUSSION THREAD

**Discussion by Maxlo24** [Project-MONAI/MONAI#2993] [Q&A] [answered]
Hi,
I'm working on a project to segment parts of a 3D medical image.
I already trained a model that is giving me the region of interest in the image.

Because I need more precision, now I would like to train an other model that will only work on specific regions ( identified by the first model ) on a much more detailed image.

The question is, for the prediction, is there a way to use **sliding_window_inference** only on wanted regions ?
The objective is that I still get an output image that have the same size as the input but that the prediction only occurs in some specific regions.

Thanks, let me know if you need further explanations.

**Comment by rijobro** [comment, marked as answer]
`sliding_window_inferer` works on a batch of data, so I can't see a general way to infer each image in the batch with different crop sizes.

It would require a bit of code to be written on your part, but could you not crop your image before passing it to the sliding window inferrer? If you wanted, you could then use the inverse transformation to restore it to its original shape (so that you can combine different regions).

**Reply by Maxlo24** [reply]
Ok sorry, I wasn't very clear.
What I meant by  "the prediction only occurs in some specific regions." Is that I keep the same crop size but it crop only in some regions of the full 3D image.

Thanks, your suggestion might work too !

**Reply by rijobro** [reply]
Ok, I understand. Thankfully, it'll be easier for you to infer batches of images if the sizes all match. However, as far as I'm aware there's no functionality in sliding window to infer a subsection. It may be worth creating a feature request, but we need to think if enough people would use this functionality to justify implementing it. My gut feeling is that it would be easier to just implement it yourself.

**Reply by Maxlo24** [reply]
Ok got it, thanks for your answer !

**Reply by rijobro** [reply]
If you use `SpatialCropd`, then you can use the `inverse` method afterwards to return the image to the original shape.
